src/TheThingsNetwork.cpp

```diff
@@ -401,13 +401,12 @@ void TheThingsNetwork::autoBaud()
 
 void TheThingsNetwork::reset(bool adr)
 {
-  if (!baudDetermined)
-  {
-    autoBaud();
-  }
-
+  autoBaud();
   size_t length = readResponse(SYS_TABLE, SYS_RESET, buffer, sizeof(buffer));
 
+  autoBaud();
+  length = readResponse(SYS_TABLE, SYS_TABLE, SYS_GET_VER, buffer, sizeof(buffer));  
```

Comment:
What I meant was that it was an arbitrary choice that was not strictly necessary, while you could have left the post autobaud verification to the library user.